### PR TITLE
Command Orders Bugfix and Minimap

### PR DESCRIPTION
**Pull request by tyeagg**
## About The Pull Request
fixes bugs with command orders arrows not appearing and runtiming every time you do an order. also adds it to minimap and makes it so rally orders are useable in overwatch computers and for ai
gif (1)

## Why It's Good For The Game
easier to tell where command wants you to go and i guess its slightly easier to herd cats as command

## Changelog
:cl:
qol: you can use rally order in overwatch computers and ai
fix: order arrows not appearing and runtiming
imageadd: defend and rally arrow pointer and minimap icons for orders
/:cl:

### CODE REVIEW

**Comment by Losenis** [comment]
Based

File: code/datums/actions/order_action.dm [hunk 1]
```diff
@@ -86,7 +86,14 @@
...
 		return
-	var/datum/atom_hud/squad/squad_hud = GLOB.huds[DATA_HUD_SQUAD_TERRAGOV]
+	var/hud_type
+	if(faction == FACTION_TERRAGOV)
+		hud_type = DATA_HUD_SQUAD_TERRAGOV
+	else if(faction == FACTION_TERRAGOV_REBEL)
+		hud_type = DATA_HUD_SQUAD_REBEL
+	else if(faction == FACTION_SOM)
+		hud_type = DATA_HUD_SQUAD_SOM
+	var/datum/atom_hud/squad/squad_hud = GLOB.huds[hud_type]
```
Comment (TiviPlus) [inline code/datums/actions/order_action.dm:96]: hud type can be null here if its not marines/rebel/som, maybe add a crash or default?

File: code/game/objects/effects/temporary_visuals/miscellaneous.dm [hunk 2]
```diff
@@ -96,17 +96,29 @@
...
+		marker_flags = MINIMAP_FLAG_MARINE_REBEL
+	else if(hud_type == DATA_HUD_SQUAD_SOM)
+		marker_flags = MINIMAP_FLAG_MARINE_SOM
 	var/datum/atom_hud/squad/squad_hud = GLOB.huds[hud_type]
 	squad_hud.add_to_hud(src)
+	SSminimaps.add_marker(src, src.z, marker_flags, icon_state_on)
 	set_visuals(faction)
 
+/obj/effect/temp_visual/order/Destroy()
+	SSminimaps.remove_marker(src)
```
Comment (TiviPlus) [inline code/game/objects/effects/temporary_visuals/miscellaneous.dm:119]: Correct me if im wrong but doesnt SSminimaps self remove on deletion anyway

File: code/game/objects/machinery/overwatch.dm [hunk 3]
```diff
@@ -46,12 +46,14 @@ GLOBAL_LIST_EMPTY(active_cas_targets)
 	var/datum/action/innate/order/retreat_order/send_retreat_order
 	///datum used when sending a defend order
 	var/datum/action/innate/order/defend_order/send_defend_order
-
+	///datum used when sending a rally order
+	var/datum/action/innate/order/rally_order/send_rally_order
 /obj/machinery/computer/camera_advanced/overwatch/Initialize()
```
Comment (TiviPlus) [inline code/game/objects/machinery/overwatch.dm:51]:
```suggestion
	var/datum/action/innate/order/rally_order/send_rally_order

/obj/machinery/computer/camera_advanced/overwatch/Initialize()
```

File: code/datums/actions/order_action.dm [hunk 4]
```diff
@@ -156,12 +163,15 @@
 	action_icon_state = "rally"
 	verb_name = "rally to"
 	arrow_type = /obj/screen/arrow/rally_order_arrow
+	visual_type = /obj/effect/temp_visual/order/rally_order
+
+/datum/action/innate/order/rally_order/personal
 	keybind_signal = COMSIG_KB_RALLYORDER
```
Comment (TiviPlus) [inline code/datums/actions/order_action.dm:169]: Can you explain this?

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Good change

File: code/datums/actions/order_action.dm (hunk 4 above)
Comment (tyeagg) [reply code/datums/actions/order_action.dm:169]: personal meaning humans with the skill level gets it and makes them say a message but overwatch computers dont need that

File: code/game/objects/effects/temporary_visuals/miscellaneous.dm (hunk 2 above)
Comment (tyeagg) [reply code/game/objects/effects/temporary_visuals/miscellaneous.dm:119]: tested and yea

**Review by BraveMole — APPROVED** [review]

**Pull request merged**